src/app/directives/clickoutside.directive.ts

```diff
@@ -0,0 +1,23 @@
+import {Directive, ElementRef, Output, EventEmitter, HostListener} from '@angular/core';
+
+@Directive({
+   selector: '[click-outside-popup]'
+})
+
+export class ClickOutsideDirective {
+   constructor(private eref : ElementRef) {
+   }
+
+   @Output()
+   public clickOutsidePopup = new EventEmitter();
+
+   @HostListener('document:click', ['$event.target'])
+   public onClick(targetElement) {
+       const clickedInsidePopup = this.eref.nativeElement.contains(targetElement);
+       const assigned_user = targetElement.classList.contains('assigned_user');
```

Comment:
@divyanshiGupta  it would be better if assigned_user is passed as a parameter to onClick. As this is a directive it needs to be as generalised as possible.

Reply:
@nimishamukherjee agreed. Done